### ISSUE TITLE
Minimize synchronously

### ISSUE THREAD

**Issue by tamtakoe** [Swaagie/minimize#61]
Minimize synchronously as `minimize.parseSync`

**Comment by Swaagie** [comment]
what are your requirements for doing it sync?

**Comment by tamtakoe** [comment]
For example, If I caches templates
```js
templates.forEach(function(template) {
    templatesCache.push(minimize.parseSync(template))
})
```
or other minor situations when it is uncomfortable to work with callbacks

**Comment by Swaagie** [comment]
Hmm i can see the usecase. However this is slightly out of minimizes' control. E.g. the DOMParser which is used by htmlparser2 is working with a callback https://github.com/fb55/domhandler#example. The only way this could work is if all 3 modules (minimize->htmlparser2->domhandler) would be working sync. Alternative solution to you caching.

```
require('async').map(templates, minimize.parse.bind(minimize), function (err, results) {
  templatesCache = results;
})
```

or similar

**Comment by tamtakoe** [comment]
I made issue https://github.com/fb55/domhandler/issues/21

https://github.com/kangax/html-minifier works synchronously, but has verbose API and has not gulp plugin.

**Comment by fb55** [comment]
@Swaagie Calling the functions of the modules is a synchronous operation. Simply process the file, then access domhandler's `dom` property. The callback is also synchronous, you could have the handler assign the value to a local variable, then return it (not a nice solution, but works as well).

**Comment by Swaagie** [comment]
@fb55 thanks for replying :) I hope I can spend a cycle on it this weekend.

**Comment by Swaagie** [comment]
Kinda fell of the radar, I'll try to spend a cycle on it asap

**Comment by manuel-di-iorio** [comment]
Looking for the sync feature :)

**Comment by Swaagie** [comment]
Finally fixed with https://github.com/Swaagie/minimize/pull/80 and released as 2.0.0
